internal/protocols/ethereum/signer.go

```diff
@@ -31,8 +32,11 @@ type SignerOptions struct {
 }
 
 // NewSigner returns a new ethereum signer that can be used to sign transactions.
-func NewSigner(privateKey crypto.PrivateKey) Signer {
-	return Signer{privateKey: privateKey}
+func NewSigner(privateKey crypto.PrivateKey) (*Signer, error) {
```

Comment:
The changes in this file look out of scope for https://github.com/mailchain/mailchain/issues/350. 
they appear related to https://github.com/mailchain/mailchain/issues/331 could you move them in to a seperate PR for that to make it clearer. That would be awesome as then its 2 PR's!!!!! 😄